hooks/useThemeSettings/utils.ts

```diff
@@ -0,0 +1,32 @@
+import type { ThemeSettings } from 'types';
+
+interface ThemeSettingsQuery extends Omit<ThemeSettings, 'show_date' | 'show_subtitle'> {
+    show_date: string;
+    show_subtitle: string;
+}
+
+export function parseQuery(query: Partial<ThemeSettingsQuery>): Partial<ThemeSettings> {
+    let show_date: boolean | undefined;
+    let show_subtitle: boolean | undefined;
+
+    try {
+        show_date = query.show_date ? JSON.parse(query.show_date) : undefined;
+        // eslint-disable-next-line no-empty
+    } catch (error) {}
+
+    try {
+        show_subtitle = query.show_subtitle ? JSON.parse(query.show_subtitle) : undefined;
+        // eslint-disable-next-line no-empty
+    } catch (error) {}
+
+    return {
+        ...(query.accent_color && { accent_color: query.accent_color }),
+        ...(query.font && { font: query.font }),
+        ...(query.header_background_color && {
+            header_background_color: query.header_background_color,
+        }),
+        ...(query.header_link_color && { header_link_color: query.header_link_color }),
+        ...(typeof show_date === 'boolean' && { show_date }),
+        ...(typeof show_subtitle === 'boolean' && { show_subtitle }),
+    };
```

Comment:
I could not think of a simpler approach here. The idea is that we don't want to set keys that are `undefined` as they would otherwise overwrite other properties.

Theoretically it's possible to use lodash's `merge` function that skips keys with `undefined` values but I didn't want to introduce another package for a pretty simple thing.

Reply:
We can do something like this:
```ts
const settings = { ... } // all of the properties, including the `undefined`
return Object.fromEntries(Object.entries(settings).filter((key, value) => typeof value !== 'undefined'));
```